tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -8978,10 +8978,9 @@ bool SPIRVEmitter::processTessellationShaderAttributes(
     theBuilder.addExecutionMode(entryFunctionId, hsExecMode, {});
   }
 
-  // Early return for domain shaders as domain shaders only takes the 'domain'
-  // attribute.
-  if (shaderModel.IsDS())
-    return true;
+  // Note: For DS, Partitioning and OutputTopology are valid in DX but not in
```

Comment:
We need to update the comment here to make it clear the differences between HLSL and GLSL:

* HLSL requires `partitioning` and `outputtopology` to be attached in HS
* GLSL requires the corresponding `layout(...)` to be declared in DS

SPIR-V is highly influenced by GLSL, so it's not surprising that drivers are expecting execution modes translated from the above high-level language features to be in DS. 

And we are explicitly allowing invalid HLSL shaders (according to the HLSL doc) for Vulkan usage.